jerry-core/ecma/builtin-objects/ecma-builtin-regexp.c

```diff
@@ -112,7 +112,20 @@ ecma_builtin_regexp_dispatch_construct (const ecma_value_t *arguments_list_p, /*
 
     if (ecma_is_value_empty (ret_value))
     {
-      ret_value = ecma_op_create_regexp_object (pattern_string_p, flags_string_p);
+      uint16_t flags = 0;
+
+      if (flags_string_p != NULL)
+      {
+        ECMA_TRY_CATCH (empty, re_parse_regexp_flags (flags_string_p, &flags), ret_value);
+        ECMA_FINALIZE (empty);
+
+        if (!ecma_is_value_empty (ret_value))
+        {
+          return ret_value;
```

Comment:
This early return here is suspicious as it will prevent `pattern_string_p` and `flags_string_p` being deref'd later on. Shouldn't this rather be something like:

```c
    uint16_t flags = 0;
    if (ecma_is_value_empty (ret_value) && (flags_string_p != NULL))
    {
      ret_value = re_parse_regexp_flags (flags_string_p, &flags);
    }

    if (ecma_is_value_empty (ret_value))
    {
      ret_value = ecma_op_create_regexp_object (pattern_string_p, flags);
    }
```

Reply:
Oh nice catch, I somehow missed that part, thank you for that!